### PR TITLE
Fix: Improve Relative Date Categorical Expressions

### PR DESCRIPTION
**Pull request by eakmanrq**
Prior to this change if I said I wanted a plan to run for 1 day ago it would effectively be a no-op since 1 day ago has time and since we don't want to load a partial we move the date up and therefore it is 0 days ago. Really though when I said "1 day ago" I didn't mean "I day ago at this same time" but rather "1 day ignoring time". Therefore we set this now to midnight and there is no longer a partial.

### CODE REVIEW

File: sqlmesh/utils/date.py [hunk 1]
```diff
@@ -279,3 +283,16 @@ def time_like_to_str(time_like: TimeLike) -> str:
...
         return to_ds(time_like)
     return to_ts(time_like)
+
+
+def is_catagorical_relative_expression(expression: str) -> bool:
+    # The Freshness Date Data Parser doesn't support plural units so we add the `s?` to the expression
+    freshness_date_parser.PATTERN = re.compile(
+        r"(\d+[.,]?\d*)\s*(%s)s?\b" % freshness_date_parser._UNITS, re.I | re.S | re.U  # type: ignore
+    )
+    if str(expression).lower() in {"today", "yesterday", "tomorrow"}:
```
Comment (izeigerman) [inline sqlmesh/utils/date.py:293]: here's another one
```
In [1]: import dateparser

In [2]: dateparser.parse("the day before yesterday")
Out[2]: datetime.datetime(2023, 10, 2, 13, 10, 10, 485354)
```

Comment (izeigerman) [reply sqlmesh/utils/date.py:293]: We should `.strip()` the string too.

File: sqlmesh/utils/date.py [hunk 2]
```diff
@@ -146,8 +158,8 @@ def to_datetime(value: TimeLike, relative_base: t.Optional[datetime] = None) ->
     if dt is None:
         raise ValueError(f"Could not convert `{value}` to datetime.")
 
-    if dt.tzinfo:
```
Comment (tobymao) [inline sqlmesh/utils/date.py:149]: revert this back

File: sqlmesh/utils/date.py [hunk 3]
```diff
@@ -279,3 +291,12 @@ def time_like_to_str(time_like: TimeLike) -> str:
     if is_date(time_like):
         return to_ds(time_like)
     return to_ts(time_like)
+
+
+def is_catagorical_relative_expression(expression: str) -> bool:
+    if str(expression).lower() in {"today", "yesterday", "tomorrow"}:
```
Comment (tobymao) [inline sqlmesh/utils/date.py:297]: you don't need to call str, it is a str

Comment (tobymao) [inline sqlmesh/utils/date.py:297]: make the set a constant

File: sqlmesh/utils/date.py [hunk 4]
```diff
@@ -279,3 +291,12 @@ def time_like_to_str(time_like: TimeLike) -> str:
...
+        return True
+    grain_kwargs = freshness_date_parser.FreshnessDateDataParser().get_kwargs(expression)
+    if not grain_kwargs:
+        return False
+    return not any(k in {"hours", "minutes", "seconds"} for k in grain_kwargs.keys())
```
Comment (tobymao) [inline sqlmesh/utils/date.py:302]: get rid of keys, make the hours a constant

File: sqlmesh/utils/date.py [hunk 5]
```diff
@@ -279,3 +291,12 @@ def time_like_to_str(time_like: TimeLike) -> str:
     if is_date(time_like):
         return to_ds(time_like)
     return to_ts(time_like)
+
+
+def is_catagorical_relative_expression(expression: str) -> bool:
+    if str(expression).lower() in {"today", "yesterday", "tomorrow"}:
+        return True
+    grain_kwargs = freshness_date_parser.FreshnessDateDataParser().get_kwargs(expression)
```
Comment (tobymao) [inline sqlmesh/utils/date.py:299]: is there state on freshnessdateparser? make that a constant too

**Review by tobymao — APPROVED** [review]

**Pull request merged**